instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java

```diff
@@ -188,14 +202,17 @@ public void addThrowable(Span span, Throwable throwable) {
   }
 
   public static <C> Context extract(C carrier, TextMapPropagator.Getter<C> getter) {
+    return extract(GlobalOpenTelemetry.getPropagators(), carrier, getter);
+  }
+
+  public static <C> Context extract(
+      ContextPropagators propagators, C carrier, TextMapPropagator.Getter<C> getter) {
```

Comment:
I'd suggest converting this to non-static so that it can use the new `propagators` instance field (I don't think it was particularly important for it to be static)
```suggestion
  public <C> Context extract(C carrier, TextMapPropagator.Getter<C> getter) {
```

Reply:
the caller didn't have access to a BaseTracer, so I didn't want to go too crazy unwinding that sweater.

Reply:
ya no worries, can you add a TODO here

```
// TODO convert to non-static, so it can use the propagators instance field
```

Reply:
Here's the troublesome usage:  https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/5f263644da9383616016d2cc007a3301f7885c4b/instrumentation/rmi/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rmi/context/server/ContextDispatcher.java#L53

Reply:
I think this could be converted to:

```
Context context = tracer().extract(payload, GETTER);
```

Reply:
I can deprecate the static method and convert all the other usages to use a non-static one. That's not a huge change, and is a good move in the right direction.